### ISSUE TITLE
maximalist CI

### ISSUE THREAD

**Issue by cfm** [freedomofpress/securedrop#6980]
## Description

After v2.8.0, the Server's steady state will increase the risk of regressions (from upstream changes, tooling changes, etc.) going unnoticed.  While we may want to add tests or CI jobs for specific things, to start we can just:
1. [ ] remove all branch filters (`i18n-*`, `stg-*`);
2. [ ] run all jobs on all branches and pull requests;
3. [ ] run all jobs in nightlies; and
4. [ ] alarm loudly on failures.

Basically: run all the tests everywhere and every time.

## Possible expansions of scope
* #6941

## User Stories

As a maintainer, I want to be able to trust that CI is a reliable and comprehensive gauge of this repository's health.

As a maintainer, I don't want to miss failing CI.
